src/lib/blkdata_svc/blkdata_service.cpp

```diff
@@ -110,6 +110,11 @@ folly::Future< std::error_code > BlkDataService::async_read(MultiBlkId const& bl
 
 folly::Future< std::error_code > BlkDataService::async_read(MultiBlkId const& blkid, sisl::sg_list& sgs, uint32_t size,
                                                             bool part_of_batch) {
+    return async_read(std::vector<MultiBlkId>{blkid}, sgs, size, part_of_batch);
```

Comment:
This will add latency to the operation purely to reduce the redundancy in the code no? Can we do decompose these two into three functions even if there is some repetition so we don't pay a cost?

Reply:
Yes, it is to purely reduce duplicate code. I can separate the code to avoid the latency